coderd/workspaces_test.go

```diff
@@ -1412,6 +1412,37 @@ func TestWorkspaceUpdateTTL(t *testing.T) {
 	})
 }
 
+func TestWorkspaceUpdateOwner(t *testing.T) {
```

Comment:
Do you see some other use cases we should cover with unit tests? I'm thinking about template access permissions.

Reply:
Absolutely. The story around permissions is something that would make sense to add, since it's currently a naive assumption that if you have access to the service you should be able to transfer ownership. In reality it should check permissions and roles and tests around this to verify would make sense.

Reply:
> In reality it should check permissions and roles and tests around this to verify would make sense.

You don't need to implement it, but could you please give some examples?